doc/stdenv/cross-compilation.chapter.md

```diff
@@ -0,0 +1,167 @@
+# Cross-compilation {#chap-cross}
+
+## Introduction {#sec-cross-intro}
+
+"Cross-compilation" means compiling a program on one machine for another type of machine. For example, a typical use of cross-compilation is to compile programs for embedded devices. These devices often don't have the computing power and memory to compile their own programs. One might think that cross-compilation is a fairly niche concern. However, there are significant advantages to rigorously distinguishing between build-time and run-time environments! Significant, because the benefits apply even when one is developing and deploying on the same machine. Nixpkgs is increasingly adopting the opinion that packages should be written with cross-compilation in mind, and nixpkgs should evaluate in a similar way (by minimizing cross-compilation-specific special cases) whether or not one is cross-compiling.
+
+This chapter will be organized in three parts. First, it will describe the basics of how to package software in a way that supports cross-compilation. Second, it will describe how to use Nixpkgs when cross-compiling. Third, it will describe the internal infrastructure supporting cross-compilation.
+
+
+## Packaging in a cross-friendly manner {#sec-cross-packaging}
+
+### Platform parameters {#ssec-cross-platform-parameters}
+
+Nixpkgs follows the [conventions of GNU autoconf](https://gcc.gnu.org/onlinedocs/gccint/Configure-Terms.html). We distinguish between 3 types of platforms when building a derivation: _build_, _host_, and _target_. In summary, _build_ is the platform on which a package is being built, _host_ is the platform on which it will run. The third attribute, _target_, is relevant only for certain specific compilers and build tools.
+
+In Nixpkgs, these three platforms are defined as attribute sets under the names `buildPlatform`, `hostPlatform`, and `targetPlatform`. They are always defined as attributes in the standard environment. That means one can access them like:
+
+```
+{ stdenv, fooDep, barDep, .. }: ...stdenv.buildPlatform...
+```
+
+`buildPlatform` The "build platform" is the platform on which a package is built. Once someone has a built package, or pre-built binary package, the build platform should not matter and can be ignored.
+
+`hostPlatform` The "host platform" is the platform on which a package will be run. This is the simplest platform to understand, but also the one with the worst name.
+
+`targetPlatform` The "target platform" attribute is, unlike the other two attributes, not actually fundamental to the process of building software. Instead, it is only relevant for compatibility with building certain specific compilers and build tools. It can be safely ignored for all other packages.
```

Comment:
I think anything in this section that was a dd/dt should be a list and have a colon and maybe also be bold. Like:

```suggestion
- *`targetPlatform`*: The "target platform" attribute is, unlike the other two attributes, not actually fundamental to the process of building software. Instead, it is only relevant for compatibility with building certain specific compilers and build tools. It can be safely ignored for all other packages.
```

Reply:
Pandoc also supports definition lists: https://pandoc.org/MANUAL.html#definition-lists